common/controllers/form-wizard.js

```diff
@@ -17,14 +18,37 @@ class FormController extends Controller {
     next()
   }
 
+  setupArbitraryMultipleFields(req, res, next) {
+    if (req.method === 'POST') {
+      const jsItemsProperty = req.body['js-items-property']
+      const jsItemsLength = req.body['js-items-length']
+
+      if (jsItemsProperty && jsItemsLength !== undefined) {
+        const values = req.sessionModel.toJSON()
+        const itemsLength = Number(jsItemsLength)
+
+        // add items
+        while (values[jsItemsProperty].length < itemsLength) {
+          values[jsItemsProperty].push({})
+        }
+
+        // remove items
+        values[jsItemsProperty].length = itemsLength
+
+        req.sessionModel.set(values)
```

Comment:
This does also feel like it's doing similar things to what the non-js version does. Is there a way we could find a way to neaten it and share the logic rather than requiring extra JS-only values to be sent in the form?

Reply:
Suggestions on a postcard, please

Reply:
Considerably rewritten  - I think it's simpler and it certainly no longer requires the js-only values